packages/lexical-markdown/src/utils.ts

```diff
@@ -460,3 +460,46 @@ export function isEmptyParagraph(node: LexicalNode): boolean {
       MARKDOWN_EMPTY_LINE_REG_EXP.test(firstChild.getTextContent()))
   );
 }
+
+export const PHONE_NUMBER_REGEX = /^\+?[0-9\s()-]{5,}$/;
+
+/**
+ * Formats a URL string by adding appropriate protocol if missing
+ *
+ * @param url - URL to format
+ * @returns Formatted URL with appropriate protocol
+ */
+export function formatUrl(url: string): string {
+  // Check if URL already has a protocol
+  if (url.match(/^[a-z][a-z0-9+.-]*:/i)) {
+    // URL already has a protocol, leave it as is
+    return url;
+  }
+  // Check if it's a relative path (starting with '/', '.', or '#')
+  else if (url.startsWith('/') || url.startsWith('.') || url.startsWith('#')) {
+    // Relative path, leave it as is
+    return url;
+  }
+
+  // Check for email address - ensure it has exactly one '@' with non-empty username and domain portions
+  else if (
+    url.includes('@') &&
+    url.split('@').length === 2 &&
+    url.split('@')[0].length > 0 &&
+    url.split('@')[1].length > 0
+  ) {
+    return `mailto:${url}`;
+  }
```

Comment:
It's inefficient to do it this way since the split is computed three times (and the first split is also equivalent to the includes). There are also esoteric cases where you *could* have more than one @ in the local part of an address if the local part is quoted. I think it's better to leave it as simply `url.includes('@')` or maybe something like `url.match(/^(?:".+"|[^@]+)@[^@]+/)` and assume that if someone is writing a link they will specify a protocol if it's not supposed to be treated as an email address. It's not like we try and validate URIs that do have a scheme nor is there really something we can do if we consider the parse to be a warning or failure.

```suggestion
  else if (url.includes('@')) {
    return `mailto:${url}`;
  }
```

See also:
* https://en.wikipedia.org/wiki/Email_address#Examples
* https://datatracker.ietf.org/doc/html/rfc2822#section-3.4.1
* https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.5

Reply:
Interesting, I didn't know we can have a quoted string in email. This does make this split invalid. 

`url.includes('@')` For checks as simple as these, it will also pass for these types of strings: `'@abcd'`, `'abcd@'` and `'@'`. 

Would this be an accepted behaviour?

Reply:
Yes, I think so. Most of those would be invalid as an https url too since @ cant be in the domain. This is just transforming the url into what is most likely to be useful, it’s not a strict validator.